editor/plugins/gdextension/cpp_scons/cpp_scons_gdext_creator.cpp

```diff
@@ -0,0 +1,232 @@
+	int result = OS::get_singleton()->execute("git", args, &output);
+	ERR_FAIL_COND_MSG(result != 0 || output.is_empty(), "Could not run git command. Please clone godot-cpp manually in order to have a working GDExtension plugin.");
+	args.front()->set("clone");
+	args.push_back("--single-branch");
+	args.push_back("--branch");
+	args.push_back(VERSION_BRANCH);
+	args.push_back("https://github.com/godotengine/godot-cpp");
+	const String godot_cpp_path = p_parent_path.trim_prefix("res://").path_join("godot-cpp");
+	args.push_back(godot_cpp_path);
+	ep.step("Cloning godot-cpp...", 1);
+	output = "";
+	result = OS::get_singleton()->execute("git", args, &output);
+	Ref<DirAccess> dir = DirAccess::create(DirAccess::ACCESS_RESOURCES);
+	if (result != 0 || !dir->dir_exists(godot_cpp_path)) {
+		args.get(3) = "master";
+		output = "";
+		result = OS::get_singleton()->execute("git", args, &output);
+	}
+	ERR_FAIL_COND_MSG(result != 0 || !dir->dir_exists(godot_cpp_path), "Failed to clone godot-cpp. Please clone godot-cpp manually in order to have a working GDExtension plugin.");
+	ep.step("Performing initial compile... (this may take several minutes)", 2);
+	if (p_compile) {
+		result = OS::get_singleton()->execute("scons", List<String>());
+		ERR_FAIL_COND_MSG(result != 0, "Failed to compile godot-cpp. Please ensure SCons is installed, then run the `scons` command in your project.");
+	}
+	ep.step("Done!", 3);
+}
+
+void CppSconsGDExtensionCreator::_make_dir_in_res(const String &p_dir_path) {
+	Ref<DirAccess> dir = DirAccess::create(DirAccess::ACCESS_RESOURCES);
+	dir->make_dir_recursive(p_dir_path);
+}
+
+String CppSconsGDExtensionCreator::_process_template(const String &p_contents) {
+	String ret;
+	if (_strip_module_defines) {
+		bool keep = true;
+		PackedStringArray lines = p_contents.split("\n");
+		for (int i = 0; i < lines.size(); i++) {
+			String line = lines[i];
+			if (line == "#if GDEXTENSION" || line == "#else") {
+				continue;
+			} else if (line == "#elif GODOT_MODULE") {
+				keep = false;
+				continue;
+			} else if (line == "#endif") {
+				keep = true;
+				continue;
+			}
+			if (keep) {
+				ret += line + "\n";
+			}
+		}
+	} else {
+		ret = p_contents;
+	}
+	ret = ret.replace("__BASE_NAME__", _base_name);
+	ret = ret.replace("__LIBRARY_NAME__", _library_name);
+	ret = ret.replace("__GODOT_VERSION__", VERSION_BRANCH);
+	ret = ret.replace("__BASE_PATH__", _res_path.trim_prefix("res://"));
+	ret = ret.replace("__UPDIR_DOTS__", _updir_dots);
+	return ret;
+}
+
+void CppSconsGDExtensionCreator::_write_file(const String &p_file_path, const String &p_contents) {
+	Error err;
+	Ref<FileAccess> file = FileAccess::open(p_file_path, FileAccess::WRITE, &err);
+	ERR_FAIL_COND_MSG(err != OK, "Couldn't write file at path: " + p_file_path + ".");
+	file->store_string(_process_template(p_contents));
+	file->close();
+}
+
+void CppSconsGDExtensionCreator::_ensure_file_contains(const String &p_file_path, const String &p_new_contents) {
```

Comment:
Shouldn't it be called `_ensure_file_contents`?

Reply:
The function ensures that the given string is present inside of the file. Meaning, the string is contained in the file.